theano/tensor/extra_ops.py

```diff
@@ -11,6 +11,56 @@
 tensor = basic
 
 
+class CpuContiguous(theano.Op):
+    """
+    Check to see if the input is c-contiguous,
+    if it is, do nothing, else return a contiguous array
+    """
+    __props__ = ()
+    view_map = {0: [0]}
+    
+    def make_node(self, x):
+        x_ = theano.tensor.as_tensor_variable(x)
+        return theano.Apply(self, [x_], [x_.type()])
+    
+    def perform(self, node, inputs, output_storage):
+        x, = inputs
+        y = output_storage[0]        
+        # if the ouput is contiguous do nothing, else copy
+        # the input
+        if not x.flags['C_CONTIGUOUS']:
+            x = x.copy()
+        assert x.flags['C_CONTIGUOUS']
+        y[0] = x 
+
+    def c_code(self, node, name, inames, onames, sub): 
+        x, = inames
+        y, = onames
+        code = """
+            if (!PyArray_CHKFLAGS(%(x)s, NPY_ARRAY_C_CONTIGUOUS)){
+                // check to see if output is contiguous first 
+                if (%(y)s != NULL && PyArray_CHKFLAGS(%(y)s, NPY_ARRAY_C_CONTIGUOUS)){
+                    PyArray_CopyInto(%(y)s, %(x)s);
+                }
+                else{
+                    Py_XDECREF(%(y)s);
+                    Py_XINCREF(%(x)s);
```

Comment:
This adds an extra useless reference to `x`.  PyArray_GETCONTIGUOUS already returns a new reference.